src/lib.rs

```diff
@@ -155,6 +155,8 @@ pub struct Package {
     pub features: HashMap<String, Vec<String>>,
     /// Path containing the `Cargo.toml`
     pub manifest_path: String,
+    /// Contents of the free form package.metadata section
+    pub metadata: Option<serde_json::Map<String, serde_json::Value>>,
```

Comment:
Does this need to be an `Option`? Can't it simply be an empty table? I don't want to distinguish between no table and an empty table. That feels like noise

Reply:
An empty table doesn't parse if the key is not present. But I pushed a new version that makes the field a `serde_json::Value` (null if not present).

Reply:
Have you tried `#[serde(default)]` on the `metadata` field?

Reply:
It seems like the metadata field is always present and `null` if not set. `serde(default)` fails in this case. So I think `serde_json::Value` is the correct type here.

Reply:
But `serde(default)` helps that no “missing field `metadata`” error occurs on stable.